operator/cmd/mesh/upgrade.go

```diff
@@ -213,7 +213,7 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l *Logger) (err error) {
 
 	// Apply the Istio Control Plane specs reading from inFilenames to the cluster
 	err = ApplyManifests(nil, args.inFilenames, args.force, rootArgs.dryRun,
-		rootArgs.verbose, args.kubeConfigPath, args.context, args.wait, upgradeWaitSecWhenApply, l)
+		rootArgs.verbose, args.kubeConfigPath, "", args.context, args.wait, upgradeWaitSecWhenApply, l)
```

Comment:
TODO: add command to allow switching from "self-generated root CA" to "intermediate CA" mode, preserving the original cluster CA key-pair and adding a cert including the cluster name and signed by an external root CA.

This should probably be done before the upgrade, part of multicluster commands.
@ayj

Reply:
Last I checked, istiod only checked for the existence of cacerts at startup. I had to restart when switching from self-signed to plugged. If the user is upgrading from self-signed to plugged with no other changes, we may need to trigger a rolling restart of istiod to pickup the new cert. Alternatively, istiod would watch for secret updates and automatically re-load the certs.

cc @myidpt